templates/CRM/common/joomla.tpl

```diff
@@ -37,7 +37,7 @@
 
     <div class="clear"></div>
 
-    {if $localTasks}
+    {if !empty($localTasks)}
```

Comment:
This isn't compatible with the escape-by-default. Our strategy is to try to make sure it's defined.

Reply:
Obviously, this is moot since the specific code-path is dead anyway. But a few observations:

* `standalone.tpl` was already doing something similar (albeit `isset()` instead of `empty()`).
* There are >500 places in `civicrm-core:/templates/` which use `empty()` guards.
* It's extremely common for templates to allow "is (not) defined" checks -- even for apps with "escape by default", even for apps that are "secure".
* I know that Civi-flavored Smarty has funny edges, but I can never remember them, because they don't really make sense. It's always a 15 minute effort to re-understand these quirks.
* I checked the "Dev Guide" and couldn't find anything relevant in "Template Reference" or "Standards". 

Suggestions:

* If basic conditionals and basic outputs from regular Smarty don't work in Civi-flavored Smarty... then we need a document which shows how to use Civi-flavored Smarty. Just a list of 5-10 examples of common idioms/tasks ("output plain text variable", "output HTML variable", "test variable content", etc). For each idiom, describe whether it's encouraged or discouraged. (If discouraged, explain why.) 
* Or else find a way to make regular Smarty work...

Reply:
I have a WIP doc which I keep not getting back to: https://lab.civicrm.org/documentation/docs/dev/-/issues/880. There is a little bit at https://docs.civicrm.org/dev/en/latest/security/outputs/#escape-by-default